File: solutions/beecrowd/2139/2139.cpp
#include <cstdint>
#include <iostream>

std::int16_t calculate_days(std::int16_t month, std::int16_t day) {
    std::int16_t month_days[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 25};
    std::int16_t total_days = 0;

    for (std::int16_t i = month; i <= 12; i++) {
        total_days += month_days[i];
    }

    return total_days - day;
}

int main() {
    std::int16_t month, day;

    while (std::cin >> month >> day) {
        if (month == 12 && day == 25) {
            std::cout << "E natal!" << std::endl;
        } else if (month == 12 && day == 24) {
            std::cout << "E vespera de natal!" << std::endl;
        } else if (month == 12 && day > 25) {
            std::cout << "Ja passou!" << std::endl;
        } else {
            std::cout << "Faltam " << calculate_days(month, day)
                      << " dias para o natal!" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/2139/2139.py
import sys

input = sys.stdin.readline
output = sys.stdout.write


def calculate_days_to_christmas(month, day):
    days_until_christmas = {
        1: 360,
        2: 329,
        3: 300,
        4: 269,
        5: 239,
        6: 208,
        7: 178,
        8: 147,
        9: 116,
        10: 86,
        11: 55,
        12: 25,
    }

    return days_until_christmas[month] - day


while line := input().strip():
    month, day = map(int, line.split())

    if month == 12:
        if day == 25:
            output('E natal!\n')
            continue
        if day == 24:
            output('E vespera de natal!\n')
            continue
        if day > 25:
            output('Ja passou!\n')
            continue

    output(
        f'Faltam {calculate_days_to_christmas(month, day)} dias para o natal!\n'
    )
